backend/db/refresh.py

```diff
@@ -64,6 +69,13 @@ def refresh_db(
         print(f"Database refresh incomplete. An exception occurred.", file=sys.stderr)
         raise err
 
+    if check_db_status_var('new_request_while_refreshing'):
+        # possibly look at the timestamp for sanity check
+        delete_db_status_var('new_request_while_refreshing')
+        refresh_db(use_local_database=use_local_database)
```

Comment:
@hugwuoke I think the way you've set this up will work. Thanks much!
It looks like this is ready for me to take it on a test run now, as the PR is not a draft. Is that correct? If so I'll make sure it works and merge~

Reply:
Yes I would say its ready !